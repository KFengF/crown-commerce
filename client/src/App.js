import React from 'react';
import { Switch, Route, BrowserRouter, Redirect } from 'react-router-dom';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import Header from './components/header/Header';
import HomePage from './pages/home-page/HomePage';
import ShopPage from './pages/shop-page/ShopPage';
import FormsPage from './pages/forms-page/FormsPage';
import CheckoutPage from './pages/checkout-page/CheckoutPage';
import { currentUserSelector } from './utils/redux/user/userSelectors';
import './App.css';

class App extends React.Component {
	/* unsubscribeFromAuth = null; */

	componentDidMount() {
		/* antes de usar saga:
		const { setCurrentUser } = this.props;
		this.unsubscribeFromAuth = auth.onAuthStateChanged(async userAuth => {
			//Este metodo invoca el callback cuando se logea o deslogea
			//La sesion del usuario persiste hasta que se deslogee

			if (userAuth) {
				//Si se logea
				const userRef = getUserDocReference(userAuth);
	
				userRef.onSnapshot(snapshot => {
					//este metodo permite usar el snapshot, y esta pendiente
					//de cambios en el snapshot
	
					setCurrentUser({
						id: snapshot.id,
						...snapshot.data()
					});
				});
			} else { //si se deslogea
				setCurrentUser(null);
			}
		}, error => console.log(error)); */

		/* La funcion para agregar los items al firebase
		addCollsAndDocs(
			'collections', 
			Object.keys(collections).map(key => ({
				title: key,
				items: collections[key].items
			}))
		); */
	}

	componentWillUnmount() {
		/* this.unsubscribeFromAuth(); */
	}

	render() {
		return (
			<BrowserRouter>
				<Header />
				<Switch>
					<Route exact path="/" component={ HomePage } />
					<Route path="/shop" component={ ShopPage } />
					<Route exact path="/signin" render={ 
						() => this.props.currentUser ? <Redirect to='/' /> : <FormsPage /> 
					} />
					<Route exact path="/checkout" component={ CheckoutPage } />
				</Switch>
			</BrowserRouter>
		);
	}
}


const mapStateToProps = createStructuredSelector({ 
	currentUser: currentUserSelector
});


/* const mapDispatchToProps = dispatch => ({
	setCurrentUser: user => dispatch(setCurrentUser(user)),
}); */

export default connect(mapStateToProps)(App);
/* connect es un high order component, devuelve un componente con 
la funcion setCurrentUser como props esta actualiza el store con 
otro setCurrentUser que esta en redux/user/userAction.js y esta 
funcion devuelve un objeto con propiedades payload y type */