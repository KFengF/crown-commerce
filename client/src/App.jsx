import React from 'react';
import { Provider } from 'react-redux';
import { PersistGate } from 'redux-persist/integration/react';
import AppRouter from './App.router';
import { store, persistor } from './utils/redux/store';

const App = () => (
    <Provider store={ store } >
        {/* Provider sirve para que toda la app pueda accder al store */}
        <PersistGate persistor={ persistor } >
            <AppRouter />
        </PersistGate>
    </Provider>
);

export default App;