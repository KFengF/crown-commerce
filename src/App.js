import React from 'react';
import { Switch, Route, BrowserRouter } from 'react-router-dom';
import { auth, createUserDoc } from './utils/firebase';
import Home from './pages/home/Home';
import Shop from './pages/shop/Shop';
import Header from './components/header/Header';
import Forms from './pages/forms/Forms';
import './App.css';

class App extends React.Component {
	constructor() {
		super();

		this.state = {
			currentUser: null
		}

		this.unsubscribeFromAuth = null;
	}

	componentDidMount() {
		this.unsubscribeFromAuth = auth.onAuthStateChanged(async userAuth => {
			//Este metodo invoca el callback cuando se logea o deslogea
			//La sesion del usuario persiste hasta que se deslogee

			if (userAuth) {
				//Si se logea
				const userRef = await createUserDoc(userAuth);
	
				userRef.onSnapshot(snapshot => {
					/* este metodo se usa para chekear si el base de datos cambio
					pero lo usamos porque cada vez que lo llamamos nos da un 
					callback con el objeto snapshot */
	
					this.setState({
						currentUser: {
							id: snapshot.id,
							...snapshot.data()
						}
					}, () => console.log(this.state));
				});
			} else { //si se deslogea
				this.setState({ currentUser: userAuth }, () => console.log(this.state))
			}
		});
	}

	componentWillUnmount() {
		this.unsubscribeFromAuth();
	}

	render() {
		return (
			<BrowserRouter>
				<Header currentUser={ this.state.currentUser } />
				<Switch>
					<Route exact path="/" component={ Home } />
					<Route path="/shop" component={ Shop } />
					<Route path="/signin" component={ Forms } />
				</Switch>
			</BrowserRouter>
		);
	}
}

export default App;