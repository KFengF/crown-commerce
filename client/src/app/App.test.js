import React from 'react';
import { shallow } from 'enzyme';
import App from './App.component';
import { AppRouter } from './App.router';

describe('App component', () => {
    it('should render App component', () => {
        expect(shallow(<App />)).toMatchSnapshot();
    });
});

describe('App router', () => {
    it('should render App router', () => {
        expect(shallow(<AppRouter />)).toMatchSnapshot();
    });
});