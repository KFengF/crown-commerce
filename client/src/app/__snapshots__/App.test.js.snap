// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should render App component 1`] = `ShallowWrapper {}`;

exports[`App router should render App router 1`] = `ShallowWrapper {}`;
